services/workflows-service/src/providers/file/file.service.ts

```diff
@@ -266,22 +272,26 @@ export class FileService {
     const remoteFileNamePrefix = fileDetails.id || getDocumentId(fileDetails, false);
 
     let tmpLocalFilePath;
+
     if (shouldDownloadFromSource) {
       const tmpFile = tmp.fileSync();
+
       tmpLocalFilePath = await sourceServiceProvider.download(sourceRemoteFileConfig, tmpFile.name);
     }
 
-    const file = await fromFile(tmpLocalFilePath ?? fileDetails.uri);
-
+    const fileType = await getFileMetadata({
+      file: tmpLocalFilePath || fileDetails.uri,
```

Comment:
why did you switch to `||` instead of `??` operator ?

Reply:
To cover empty strings. As for `0` or `false` its similar reasoning to empty string. Currently we need stricter runtime validation and types to truly use `??` over `||` safely.